boot/seal.go

```diff
@@ -241,17 +257,43 @@ func sealKeyToModeenvUsingSecboot(key, saveKey keys.EncryptionKey, modeenv *Mode
 		return fmt.Errorf("cannot generate key for signing dynamic authorization policies: %v", err)
 	}
 
+	runObjectKeyPCRHandle := uint32(secboot.RunObjectPCRPolicyCounterHandle)
+	fallbackObjectKeyPCRHandle := uint32(secboot.FallbackObjectPCRPolicyCounterHandle)
+	if flags.FactoryReset {
+		// during factory reset we may need to rotate the PCR handles,
+		// seal the new keys using a new set of handles such that the
+		// old sealed ubuntu-save key is still usable
```

Comment:
in practice we end up using the two set of handles in a round-robin fashion? is not super clear from the comment but I suppose that's what the code does

Reply:
extended the comment a little to give a better idea of what happens now and during first boot